servicetalk-loadbalancer-experimental-provider/src/main/java/io/servicetalk/loadbalancer/experimental/DefaultLoadBalancerProviderConfig.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer.experimental;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+import io.servicetalk.loadbalancer.LoadBalancingPolicy;
+import io.servicetalk.loadbalancer.OutlierDetectorConfig;
+import io.servicetalk.loadbalancer.P2CLoadBalancingPolicy;
+import io.servicetalk.loadbalancer.RoundRobinLoadBalancingPolicy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+
+import static java.time.Duration.ZERO;
+import static java.time.Duration.ofMillis;
+import static java.time.Duration.ofSeconds;
+import static java.util.Objects.requireNonNull;
+
+final class DefaultLoadBalancerProviderConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultHttpLoadBalancerProvider.class);
+
+    private enum LBPolicy {
+        P2C,
+        RoundRobin
+    }
+
+    // This prefix should be applied to all individual properties.
+    private static final String PROPERTY_PREFIX = "io.servicetalk.loadbalancer.experimental.";
+
+    private static final String PROP_CLIENTS_ENABLED_FOR = "clientsEnabledFor";
+
+    private static final String PROP_FAILED_CONNECTIONS_THRESHOLD = "healthCheckFailedConnectionsThreshold";
+    private static final String PROP_LOAD_BALANCING_POLICY = "clientExperimentalLoadBalancerPolicy";
+    private static final String PROP_EWMA_HALF_LIFE_MS = "experimentalLoadBalancerEwmaHalfLifeMs";
+    private static final String PROP_CONSECUTIVE_5XX = "experimentalLoadBalancerConsecutive5xx";
+    private static final String PROP_INTERVAL_MS = "experimentalLoadBalancerIntervalMs";
```

Comment:
since the `PROPERTY_PREFIX` contains loadbalancer and experimental in the package, I wonder if we could just drop the `experimentalLoadBalancer` for each property name?

Reply:
Good point, but after I move the provider back to the same module and thus the `io.servicetalk.loadbalancer` package I think it will make sense to keep them.